src/MediaWiki/Search/SearchResultSet.php

```diff
@@ -174,24 +174,10 @@ public function getTotalHits() {
 	 * Return an array of regular expression fragments for matching
 	 * the search terms as parsed by the engine in a text extract.
 	 *
-	 * This is a temporary hack for MW versions that can not cope
-	 * with no search term being returned (<1.24).
-	 *
-	 * @deprecated remove once min supported MW version has \SearchHighlighter::highlightNone()
-	 *
 	 * @return string[]
 	 */
 	public function termMatches() {
-		if ( ( $tokens = $this->getTokens() ) !== [] ) {
-			return $tokens;
-		}
-
-		if ( method_exists( '\SearchHighlighter', 'highlightNone' ) ) {
-			return [];
-		}
-
-		// Will cause the highlighter to match every line start, thus returning the first few lines of found pages.
-		return [ '^' ];
```

Comment:
If I understand it correctly, when `$this->getTokens()` returns `[]`, this function should return `[ '^' ]`.

Reply:
I'm not sure because you then have the if method exist check above which will return true in MW 1.39+.

Reply:
In that case should all calls to this function just point to `getTokens` directly?

Reply:
I'm not sure what you mean? This function is extending the MW class one. Removing this function causes the tests to fail and I'm not sure how to get it to not use this one.

Reply:
I missed that it is extending the MW class so you're right!